app/models/spree/stock_item.rb

```diff
@@ -10,6 +10,7 @@ class StockItem < ActiveRecord::Base
     # rubocop:disable Rails/UniqueValidationWithoutIndex
     validates :variant_id, uniqueness: { scope: :stock_location_id }
     # rubocop:enable Rails/UniqueValidationWithoutIndex
+    validates :count_on_hand, numericality: { greater_than_or_equal_to: 0, unless: :backorderable? }
```

Comment:
This will make the checkout fail. Did you test what the user would see in this scenario?

Reply:
why is that?

Reply:
as far as I can see, this should never happen, if it happens and if it blows up we will see it in bugsnag.